File: Sources/Server/fairy-teller/src/main/java/com/rad/fairyteller/repository/PurchaseRepository.java
package com.rad.fairyteller.repository;

import com.rad.fairyteller.domain.Purchase;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PurchaseRepository extends JpaRepository<Purchase, Long> {
    List<Purchase> findAllByUserId(Long userId);
}

File: Sources/Server/fairy-teller/src/main/java/com/rad/fairyteller/repository/PurchaseRepository.kt
package com.rad.fairyteller.repository

import com.rad.fairyteller.domain.purchase.Purchase
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PurchaseRepository : JpaRepository<Purchase, Long> {
    fun findAllByUserId(userId: Long?): List<Purchase>
}